captum/attr/_core/lime.py

```diff
@@ -0,0 +1,1007 @@
+    model.
+
+    LimeBase allows sampling points in either the interpretable space or the original
+    input space to train the surrogate model. The interpretable space is a feature
+    vector used to train the surrogate interpretable model; this feature space is often
+    of smaller dimensionality than the original feature space in order for the surrogate
+    model to be more interpretable.
+
+    If sampling in the interpretable space, a transformation function must be provided
+    to define how a vector sampled in the interpretable space can be transformed into
+    an example in the original input space. If sampling in the original input space, a
+    transformation function must be provided to define how the input can be transformed
+    into its interpretable vector representation.
+
+    More details regarding LIME can be found in the original paper:
+    https://arxiv.org/abs/1602.04938
+    """
+
+    def __init__(
+        self,
+        forward_func: Callable,
+        train_interpretable_model_func: Callable,
+        similarity_func: Callable,
+        sampling_func: Callable,
+        sample_interpretable_space: bool,
+        from_interp_rep_transform: Optional[Callable],
+        to_interp_rep_transform: Optional[Callable],
+    ) -> None:
+        r"""
+
+        Args:
+
+
+            forward_func (callable):  The forward function of the model or any
+                    modification of it
+            train_interpretable_model_func (callable): Function which trains
+                    an interpretable model and returns some representation of the
+                    interpretable model. The return type of this will match the
+                    returned type when calling attribute.
+                    The expected signature of this callable is:
+
+                    train_interpretable_model_func(
+                        interp_inputs: Tensor [2D num_samples x num_interp_features],
+                        exp_outputs: Tensor [1D num_samples],
+                        weights: Tensor [1D num_samples]
+                        **kwargs: Any
+                    ) -> Any (Representation of interpretable model)
+
+                    Note that kwargs include all kwargs passed to the attribute
+                    method.
+            similarity_func (callable): Function which takes a single sample
+                    along with its corresponding interpretable representation
+                    and returns the weight of the interpretable sample for
+                    training interpretable model. Weight is generally
+                    determined based on similarity to the original input.
+                    The original paper refers to this as a similarity kernel.
+
+                    The expected signature of this callable is:
+
+                    similarity_func(
+                        original_input: Tensor or tuple of Tensors,
```

Comment:
Sure, that's reasonable. I originally didn't since it usually may be some fixed baseline (e.g. all ones), but it might be good for consistency, will add it.

Reply:
Actually, we don't always have the original_interpretable_input, since if the user is perturbing / sampling in the interpretable space, then we don't have the transformation from original input to interpretable. I'll leave it out, but a user can pass it through kwargs if needed.